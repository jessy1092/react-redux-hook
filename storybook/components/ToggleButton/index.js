import React from 'react';

import { useRedux } from '../../../src';

const ToggleButton = () => {
	const [state, dispatch] = useRedux();

	return (
		<button onClick={() => dispatch({ type: 'TOGGLE' })}>
			{state.toggle ? 'Click to close' : 'Click to open'}
		</button>
	);
};

export default ToggleButton;
