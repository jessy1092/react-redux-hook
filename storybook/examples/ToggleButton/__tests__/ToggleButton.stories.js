import React from 'react';
import { withKnobs } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';

import ToggleButton from '../index.js';

const stories = storiesOf('ToggleButton', module);

stories.addDecorator(withKnobs);

stories.add('__interactive', () => <ToggleButton />);
