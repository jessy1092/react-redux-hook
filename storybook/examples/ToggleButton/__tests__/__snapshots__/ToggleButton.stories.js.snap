// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<Contain(ToggleButton)
  closeTitle="Close"
  onClose={[Function]}
  onOpen={[Function]}
  openTitle="Open"
>
  <ToggleButton
    closeTitle="Close"
    handleClick={[Function]}
    openTitle="Open"
    toggle={true}
  >
    <button
      onClick={[Function]}
    >
      Close
    </button>
  </ToggleButton>
</Contain(ToggleButton)>
`;
